owlkettle/widgets.nim

```diff
@@ -504,6 +504,46 @@ renderable Label of BaseWidget:
       text = "<b>Bold</b>, <i>Italic</i>, <span font=\"20\">Font Size</span>"
       useMarkup = true
 
+renderable EditableLabel of BaseWidget:
+  editing: bool = false ## Determines whether the edit view (editing = false) or the "read" view (editing = true) is being shown
+  text: string = ""
+  
+  proc changed(text: string) ## Fired every time `text` changes.
+  proc editStateChanged(newEditState: bool) ## Fired every time `editing` changes.
+  
+  hooks:
+    beforeBuild:
+      state.internalWidget = gtk_editable_label_new("".cstring)
+    connectEvents:
+      proc changedCallback(widget: GtkWidget, data: ptr EventObj[proc (text: string)]) {.cdecl.} =
+        let text = $gtk_editable_get_text(widget)
+        EditableLabelState(data[].widget).text = text
+        data[].callback(text)
+        data[].redraw()
+      
+      proc editedCallback(widget: GtkWidget, pspec: GtkParamSpec, data: ptr EventObj[proc (newEditState: bool)]){.cdecl.} =
+        let isEditing = gtk_editable_label_get_editing(widget).bool
+        EditableLabelState(data[].widget).editing = isEditing
+        data[].callback(isEditing)
+        
```

Comment:
Missing call to redraw

Reply:
We have the same problem with `CustomWidget` that basically just redraws every time the user moves their mouse / presses any button (while inside the widget). At least for me this currently does not cause considerable issues.

Reply:
Ah, then you caught why I intentionally left the redraw out of that thing. If it is of no concern I'm happy to add it back in.